olp-cpp-sdk-dataservice-read/src/generated/api/StreamApi.cpp

```diff
@@ -217,12 +220,30 @@ Response<int> StreamApi::HandleOffsets(
   OLP_SDK_LOG_DEBUG_F(kLogTag, "handleOffsets, uri=%s, status=%d",
                       metadata_uri.c_str(), http_response.status);
 
-  // TODO: Set x_correlation_id to the value received in http_response.header
-  // when http_response.header will be implemented.
+  if (!HandleCorrelationId(http_response.headers, x_correlation_id)) {
+    OLP_SDK_LOG_INFO(kLogTag,
+                     "X-Correlation-Id was not found in responce headers");
+  }
 
   return http_response.status;
 }
 
+bool StreamApi::HandleCorrelationId(const http::HeadersType& headers,
+                                    std::string& x_correlation_id) {
+  auto it = std::find_if(std::begin(headers), std::end(headers),
+                         [](const std::pair<std::string, std::string>& header) {
```

Comment:
```suggestion
                         [](const headers::value_type& header) {
```
Maybe also a alias for the value_type should be added to NetworkTypes.h